blockdag/dag.go

```diff
@@ -1839,7 +1839,9 @@ func (dag *BlockDAG) ProcessDelayedBlocks() error {
 		earliestDelayedBlockProcessTime := dag.delayedBlocksQueue.peek().processTime
 		if dag.timeSource.AdjustedTime().After(earliestDelayedBlockProcessTime) {
 			delayedBlock := dag.popDelayedBlock()
-			err := dag.maybeAcceptBlock(delayedBlock.block, BFAfterDelay)
+			dag.dagLock.Unlock()
```

Comment:
I'd rather if you created function processBlockNoLock that has all the logic from process block, and than call it from here and the public ProcessBlock.

Unlocking here means someone might grab the process and do stuff that you don't want at this stage.